include/RAJA/pattern/launch/launch_core.hpp

```diff
@@ -208,6 +210,26 @@ class LaunchContext
     return static_cast<T*>(mem_ptr);
   }
 
+  template<typename T>
+  RAJA_HOST_DEVICE createMemoryArena(T *ptr, size_t bytes)
+  {
+    void * mem_arena_ptr = static_cast<char*>(&ptr[0]);
+  }
+
+  // TODO handle alignment
+  template<typename T>
+  RAJA_HOST_DEVICE T* getArenaMemory(size_t bytes)
+  {
+
+    // Calculate offset in bytes with a char pointer
+    void* mem_ptr = static_cast<char*>(arena_mem_ptr) + shared_arena_offset;
```

Comment:
Shouldn't it be this instead?
```suggestion
    void* mem_ptr = static_cast<char*>(mem_arena_ptr) + mem_arena_offset;
```